baseline/tf/seq2seq/training/eager.py

```diff
@@ -13,24 +13,42 @@
 from baseline.tf.seq2seq.training.utils import to_tensors, SHUF_BUF_SZ, NUM_PREFETCH
 
 
-def loss(model, features, labels):
-    # Claims its T, B, H
-    logits = tf.transpose(model(features), [1, 0, 2])
-    # So ok, then transpose this too
-    labels = tf.transpose(labels, [1, 0])
-    # TxB loss mask
-    label_lengths = features['tgt_len']
-    mx_seq_len = tf.reduce_max(label_lengths)-1
-    labels = labels[1:mx_seq_len + 1, :]
-    logits = logits[:mx_seq_len, :, :]
-    losses = tf.nn.sparse_softmax_cross_entropy_with_logits(logits=logits, labels=labels)
-    loss_mask = tf.cast(tf.sequence_mask(label_lengths-1), dtype=tf.float32)
-    losses = losses * tf.transpose(loss_mask, [1, 0])
-
-    losses = tf.reduce_sum(losses)
-    losses /= tf.cast(tf.reduce_sum(label_lengths), tf.float32)
-    return losses
+class Seq2SeqLoss(tf.keras.layers.Layer):
```

Comment:
It looks like right now the distributed mode hasnt been updated to use this loss but it should be

Reply:
We can add this later. I’ll merge for now to keep it moving